src/windows/disk.rs

```diff
@@ -383,6 +374,63 @@ fn os_string_from_zero_terminated(name: &[u16]) -> OsString {
     OsString::from_wide(&name[..len])
 }
 
+fn get_disk_kind(device_path: &[u16], borrowed_handle: &Option<HandleWrapper>) -> DiskKind {
+    let binding = (
+        borrowed_handle,
+        if borrowed_handle.is_none() {
+            unsafe { HandleWrapper::new_from_file(device_path, Default::default()) }
+        } else {
+            None
+        },
+    );
+    let handle = match binding {
+        (Some(ref handle), _) => handle,
+        (_, Some(ref handle)) => handle,
+        (None, None) => return DiskKind::Unknown(-1),
+    };
```

Comment:
```suggestion
    let Some(handle) = borrowed_handle.unwrap_or_else(|| unsafe { HandleWrapper::new_from_file(device_path, Default::default()) }) else {
        return DiskKind::Unknown(-1);
    };
```

Reply:
If there's a way to do this without an explicit binding, I'd love to know it. But the suggested approach (and similar variations I've tried) all end in `temporary value dropped while borrowed` errors.